src/e2e-test/java/io/cdap/plugin/odp/stepsdesign/RuntimeODP.java

```diff
@@ -0,0 +1,422 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.odp.stepsdesign;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.adapter.connector.SAPAdapterImpl;
+import com.google.adapter.connector.SAPProperties;
+import com.google.adapter.exceptions.SystemException;
+import com.google.adapter.util.ErrorCapture;
+import com.google.adapter.util.ExceptionUtils;
+import com.sap.conn.jco.JCoException;
+import io.cdap.e2e.pages.actions.CdfBigQueryPropertiesActions;
+import io.cdap.e2e.pages.actions.CdfGcsActions;
+import io.cdap.e2e.pages.actions.CdfLogActions;
+import io.cdap.e2e.pages.actions.CdfPipelineRunAction;
+import io.cdap.e2e.pages.actions.CdfStudioActions;
+import io.cdap.e2e.pages.locators.CdfBigQueryPropertiesLocators;
+import io.cdap.e2e.pages.locators.CdfPipelineRunLocators;
+import io.cdap.e2e.pages.locators.CdfStudioLocators;
+import io.cdap.e2e.utils.CdfHelper;
+import io.cdap.e2e.utils.GcpClient;
+import io.cdap.e2e.utils.SeleniumDriver;
+import io.cdap.e2e.utils.SeleniumHelper;
+import io.cdap.plugin.odp.actions.ODPActions;
+import io.cdap.plugin.odp.locators.ODPLocators;
+import io.cdap.plugin.odp.utils.CDAPUtils;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.log4j.Logger;
+import org.junit.Assert;
+import org.openqa.selenium.By;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import stepsdesign.BeforeActions;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+
+
+/**
+ * RuntimeODP.
+ */
+public class RuntimeODP implements CdfHelper {
+  private static final Logger logger = Logger.getLogger(RuntimeODP.class);
+  GcpClient gcpClient = new GcpClient();
+  private SAPProperties sapProps;
+  private ErrorCapture errorCapture;
+  private SAPAdapterImpl sapAdapterImpl;
+  private ExceptionUtils exceptionUtils;
+  static int dsRecordsCount;
+  static int countRecords;
+  static int presentRecords;
+  static int counter = 0;
+  static String datasource;
+  static String number;
+  static String action;
+  static int noOfRecords;
+  static String deltaLog;
+  static String rawLog;
+  static PrintWriter out;
+  private List<String> numList = new ArrayList<>();
```

Comment:
what is stored in numList? maybe use a more descriptive name?

Reply:
done